csrc/root_domain_map.cpp

```diff
@@ -227,32 +227,39 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   }
 
   if (SdpaFwdOp* op = dynamic_cast<SdpaFwdOp*>(consumer_tv_->definition())) {
+    // Note: Explicit handling of DIDx(D) until
+    // https://github.com/NVIDIA/Fuser/issues/2563 is resolved. 
     // Producers:
-    //   query = [N, H, L, E]
-    //   key = [N, H, S, E]
-    //   value = [N, H, S, Ev]
+    //   query = [DIDx(D)?, N, H, L, E]
+    //   key = [DIDx(D)?, N, H, S, E]
+    //   value = [DIDx(D)?, N, H, S, Ev]
     // Consumers:
-    //   output = [N, H, L, Ev]
-    //   logsumexp = [N, H, L]
+    //   output = [DIDx(D)?, N, H, L, Ev]
+    //   logsumexp = [DIDx(D)?, N, H, L]
     //   cum_seq_q/k = [N + 1]
 
+    unsigned int num_device_dim = producer_logical.at(0)->isDeviceDim() ? 1 : 0;
```

Comment:
```suggestion
    size_t num_device_dim = producer_logical.at(0)->isDeviceDim() ? 1 : 0;
```

It's better for num_device_dim and idx to have the same type so comparisons are unambiguous.